roles/jd-server/src/main.rs

```diff
@@ -115,43 +121,63 @@ async fn main() {
         std::time::Instant::now().sub(std::time::Duration::from_secs(60));
 
     if url.contains("http") {
-        let sender_clone = sender.clone();
+        let sender_update_mempool = sender.clone();
         task::spawn(async move {
             loop {
-                let updated_mempool =
+                let result: Result<(), mempool::error::JdsMempoolError> =
                     mempool::JDsMempool::update_mempool(mempool_cloned_.clone()).await;
-                if let Err(err) = updated_mempool {
+                if let Err(err) = result {
                     match err {
-                        mempool::JdsMempoolError::EmptyMempool => {
+                        JdsMempoolError::EmptyMempool => {
                             if last_empty_mempool_warning.elapsed().as_secs() >= 60 {
                                 warn!("{:?}", err);
                                 warn!("Template Provider is running, but its mempool is empty (possible reasons: you're testing in testnet, signet, or regtest)");
                                 last_empty_mempool_warning = std::time::Instant::now();
                             }
                         }
-                        mempool::JdsMempoolError::NoClient => {
-                            error!("{:?}", err);
-                            error!("Unable to establish RPC connection with Template Provider (possible reasons: not fully synced, down)");
-                            handle_result!(sender_clone, Err(err));
-                        }
-                        mempool::JdsMempoolError::BitcoinCoreRpcError(_) => {
-                            error!("{:?}", err);
-                            error!("Unable to establish RPC connection with Template Provider (possible reasons: not fully synced, down)");
-                            handle_result!(sender_clone, Err(err));
+                        _ => {
```

Comment:
better have a branch for each error instead of _ if you have more error that should do the same thing you can use Error::One | Error::Two => {}

Reply:
This have been addressed, please check if it looks fine

Reply:
done